starfyre/__main__.py

```diff
@@ -14,42 +16,78 @@ def write_js_file(path):
     js_store = pkg_resources.path("starfyre.js", "store.js")
     shutil.copy(str(js_store), path + "/dist/store.js")
 
-
-def create_main_file(path):
-    """
-    Creates a main file in the build directory.
-    
-    This file will be used to run the project.
-    Every starfyre build will have an `__init__.py` file in the build directory.
-    And the `__init__.py` file will have a component that will render the root component. It will be named `app`.
-
-    You can have a look at the `test-application/build/__init__.py` file to see what it looks like.
-
-    The main file is also responsible for adding the `store.js` file to the `index.html` file.
-    """
+def build_routes_output(generated_routes, path):
     output_file_path = path + "/build/__main__.py"
     write_js_file(path)
 
     with open(output_file_path, "w") as f:
         f.write(
-            """
-from . import app
+            f'''from starfyre import create_component, render_root
+from starfyre.exceptions import IndexFileConflictError
+
 import os
+import sys
 from pathlib import Path
+import importlib
 
+def generate_pages(generated_routes, path):
+    """
+    Generate HTML pages for each route in the provided list of routes.
+
+    Parameters:
+    - generated_routes (list): List of generated routes.
+    - path (str): Path to the project directory.
+
+    This function generates HTML pages for each route provided in the `generated_routes` list.
+    It imports the necessary components for each route, renders them using Starfyre,
+    and writes the rendered content to corresponding HTML files.
+    """
+    user_routes = generated_routes[:]
+
+    # index.html is to be generated first since it is the entry point of the app
+    user_routes.insert(0, 'app')
+
+    print(f"Generated routes: {{user_routes}}")
+
+    out_dir = Path(path + "/dist").resolve()
+    root = Path(out_dir / "..").resolve()
+
+    # get the user defined project name
+    app_name = (str(root).split('/'))[-1]
+
+    for route_name in user_routes:
+        print(f'route name is = {{route_name}}')
+        
+        if route_name.lower() == 'app':
+            component_key = 'app'  # For the 'app' component in `build/__init__.py`
+        else:
+            component_key = route_name
+        
+        if route_name == 'app':
+            module_name = f"{{Path(app_name)}}.build"
+        else:
+            module_name= f"{{Path(app_name)}}.build.{{route_name}}"
+        
+        try:
+            module = importlib.import_module(module_name)
+            print(f"Module is = {{module}}")
+            component = module.__dict__[component_key]
+            result = str(render_root(component)) if not route_name == 'app' else component
```

Comment:
Wait, why call render root again? I thought this was already being generated at every page?

Also is there an alternative of using `__dict__`?